app/controllers/v0/sessions_controller.rb

```diff
@@ -35,11 +35,16 @@ def saml_callback
 
     def persist_session_and_user!
       @session = Session.new(user_attributes.slice(:uuid))
-      @current_user = User.find(@session.uuid) || create_new_user
+      @current_user = User.find(@session.uuid)
+      @current_user = saml_user if @current_user.nil? || up_level?
```

Comment:
Is it possible for them to go back down a level? Are there other things that could change in the saml token other than the LOA? It seems almost like it should be:

`@current_user = saml_user if @current_user.in_any_way_different_from?(saml_user)`